src/output.cpp

```diff
@@ -2341,6 +2341,114 @@ int msgtype_to_tilecolor( const game_message_type type, const bool bOldMsg )
     return -1;
 }
 
+bool wildcard_match(const std::string &sTextIn, const std::string &sPatternIn)
+{
+    std::string sText = sTextIn;
+    std::string sPattern = sPatternIn;
+
+    //case insenitive search
+
+    /* Possible patterns
+    *
+    wooD
+    wood*
+    *wood
+    Wood*aRrOW
+    wood*arrow*
+    *wood*arrow
+    *wood*hard* *x*y*z*arrow*
+    */
+
+    if (sText == "") {
+        return false;
+    } else if (sText == "*") {
+        return true;
+    }
+
+    int iPos;
+    std::vector<std::string> vPattern;
+
+    sPattern = wildcard_trim_rule(sPattern);
+
+    wildcard_split(sPattern, '*', vPattern);
+    size_t iNum = vPattern.size();
+
+    if (iNum == 0) { //should never happen
+        return false;
+    } else if (iNum == 1) { // no * found
+        return (sText.length() == vPattern[0].length() && ci_find_substr(sText, vPattern[0]) != -1);
+    }
+
+    for (auto it = vPattern.begin(); it != vPattern.end(); ++it) {
+        if (it == vPattern.begin() && *it != "") { //beginning: ^vPat[i]
+            if (sText.length() < it->length() ||
+                ci_find_substr(sText.substr(0, it->length()), *it) == -1) {
+                return false;
+            }
+
+            sText = sText.substr(it->length(), sText.length() - it->length());
+        } else if (it == vPattern.end() - 1 && *it != "") { //linenend: vPat[i]$
+            if (sText.length() < it->length() ||
+                ci_find_substr(sText.substr(sText.length() - it->length(),
+                                            it->length()), *it) == -1) {
+                return false;
+            }
+        } else { //inbetween: vPat[i]
+            if (*it != "") {
+                if ((iPos = (int)ci_find_substr(sText, *it)) == -1) {
```

Comment:
Also in the same line: assignment in `if` is bad practice and saves only one line.